src/dataframe/io.jl

```diff
@@ -930,6 +930,178 @@ function readtable(pathname::AbstractString;
                      normalizenames = normalizenames)
 end
 
+"""
+# `inlinetable`
+A helper function to process strings as tabular data for nonstandard string
+literals.
+
+## Usage
+```julia
+inlinetable(s::AbstractString; args...) -> DataFrame
+inlinetable(s::AbstractString, flags::AbstractString; args...) -> DataFrame
+```
+
+- `s` is a string containing delimiter-separated tabular data, e.g., comma-
+  separated values
+- `args` is passed through as a named argument list to `readtable`
+- `flags` is a string containing a set of flag characters, which, if present,
+  are equivalent to supplying named arguments to `readtable` as follows:
+  - `f`: `makefactors=true`, convert string columns to `PooledData` columns
+  - `c`: `allowcomments=true`, ignore lines beginning with `#`
+  - `H`: `header=false`, do not interpret the first line as column names
+
+## Implementation notes
```

Comment:
I appreciate the intent, but in this case I don't think implementation notes are really needed. This is an internal function, and the implementation is available right below. That said, now that you've written it... :-)